### PR TITLE
XWIKI-19667: R140300001XWIKI19571DataMigration fails on postgres 12 db

### PR DESCRIPTION
**Pull request by manuelleduc**
Explicitly set the wiki id in the hibernate store during the migration to make sure to use the right database schema

Jira: https://jira.xwiki.org/browse/XWIKI-19677

### CODE REVIEW

File: xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-api/src/main/java/org/xwiki/index/internal/DefaultTasksManager.java [hunk 1]
```diff
@@ -159,6 +148,7 @@ public void initialize()
             MBEAN_NAME);
         this.queue = new PriorityBlockingQueue<>(11, Comparator.comparingLong(TaskData::getTimestamp));
         this.latestTimestampTasksMap = new ConcurrentHashMap<>();
+        startThread();
```
Comment (manuelleduc) [inline xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-api/src/main/java/org/xwiki/index/internal/DefaultTasksManager.java:151]: The thread is now started during the initialization and the tasks queue when the wikis are ready (and migrated).

**Comment by manuelleduc** [comment]
Note: I'm running manual upgrades on postgres, mysql, mariadb, oracle, and hsqldb in parallel.

File: xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-api/src/main/java/org/xwiki/index/migration/R140300001XWIKI19571DataMigration.java [hunk 2]
```diff
@@ -88,7 +88,9 @@ public XWikiDBVersion getVersion()
     public String getPreHibernateLiquibaseChangeLog() throws DataMigrationException
     {
         SessionFactory sessionFactory = this.hibernateStore.getSessionFactory();
+        String wikiId = getXWikiContext().getWikiId();
         try (SessionImplementor session = (SessionImplementor) sessionFactory.openSession()) {
+            this.hibernateStore.setWiki(session, wikiId);
```
Comment (tmortagne) [inline xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-api/src/main/java/org/xwiki/index/migration/R140300001XWIKI19571DataMigration.java:93]: Why not simply `this.hibernateStore.setWiki(session)` ?

Comment (manuelleduc) [reply xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-api/src/main/java/org/xwiki/index/migration/R140300001XWIKI19571DataMigration.java:93]: That session is already initialized with the right wiki?

Comment (manuelleduc) [reply xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-api/src/main/java/org/xwiki/index/migration/R140300001XWIKI19571DataMigration.java:93]: Ok, it's computed internally inside `setWiki(session)`

**Review by tmortagne — APPROVED** [review]

**Comment by manuelleduc** [comment]
> Note: I'm running manual upgrades on postgres, mysql, mariadb, oracle, and hsqldb in parallel.

All good except oracle where I have a technical issue preventing the test of the migration with a sub-wiki.

**Comment by manuelleduc** [comment]
> All good except oracle where I have a technical issue preventing the test of the migration with a sub-wiki.

oracle + upgrade test ok locally

**Pull request merged**